pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -766,13 +766,21 @@ func buildContainerMapFromRuntime(runtimeService internalapi.RuntimeService) (co
 
 	containerMap := containermap.NewContainerMap()
 	containerList, _ := runtimeService.ListContainers(nil)
+
+	orphanContainerIdList := make([]string, 0)
+
 	for _, c := range containerList {
 		if _, exists := podSandboxMap[c.PodSandboxId]; !exists {
-			return nil, fmt.Errorf("no PodsandBox found with Id '%s'", c.PodSandboxId)
+			klog.Warningf("no PodsandBox found with PodSandboxId: %s, containerName: %s, containerId: %s", c.PodSandboxId, c.Metadata.Name, c.Id)
+			orphanContainerIdList = append(orphanContainerIdList, c.Id)
+			continue
 		}
 		containerMap.Add(podSandboxMap[c.PodSandboxId], c.Metadata.Name, c.Id)
 	}
 
+	if len(orphanContainerIdList) > 0 {
+		return containerMap, fmt.Errorf("no PodsandBox found for this containers: '%v'", orphanContainerIdList)
+	}
 	return containerMap, nil
```

Comment:
and finally we have a curated list of valid containers, as you said old ones are bound to disappear via GC